synapseclient/client.py

```diff
@@ -266,12 +276,32 @@ def __init__(
                                 when making http requests.
             cache_root_dir:     Root directory for storing cache data.
             silent:             Suppresses message.
+            requests_session_async_synapse: The HTTPX Async client for interacting with
+                Synapse services.
+            requests_session_async_storage: The HTTPX Async client for interacting with
+                storage providers like AWS S3 and Google Cloud.
 
         Raises:
             ValueError: Warn for non-boolean debug value.
         """
         self._requests_session = requests_session or requests.Session()
 
+        httpx_timeout = httpx.Timeout(70)
+        self._requests_session_async_synapse = (
+            requests_session_async_synapse
+            or httpx.AsyncClient(
+                limits=httpx.Limits(max_connections=25),
```

Comment:
Is this a number you've experimented with and found that it was optimal? I'm seeing the default value for this arg is 100 so I wonder why it was brought down to 25

Reply:
Yes. I did some testing around this and found that lower than 25 impacted performance, higher than 25 didn't have much affect on performance - But had higher HTTP 429 rate limit errors coming back from Synapse.

Reply:
I see - and this number is the total number of connections to the API a user can make in 1 session, correct? My initial worry was that this cap might be too low for use-cases like the Challenge submissions Nextflow workflows (I'm thinking of PEGS & Dynamic Challenge), but each parallel process that uses the python client runs its own session for the most part, so perhaps this wouldn't be a problem if my understanding of what `max_connections` is is correct.

Reply:
Yes, that's correct. Each machine running this code will have this limit - and it is not shared as there isn't a way to know.

On the other hand, Synapse will kick back HTTP 429 errors if the concurrent connections are too many (Either by a single client, or a single user across many clients). This was the motivation for changing the retry logic to be time-based.

Reply:
I see, that makes sense, thanks for explaining this